### PR TITLE
Bump to newer PHPUnit

### PR DESCRIPTION
**Pull request by lornajane**
We're wildly out of date with our PHPUnit versions, which isn't an emergency but isn't great practice either. I've started by bumping the version from 5.7 to 6 (I have 6.5 locally but I don't think it's critical) and fixing all the tests that needed attention. This makes us a mere 3 years behind in PHPUnit versions :)

### CODE REVIEW

**Comment by lornajane** [comment]
Build failing because the travis file still has PHP 5 in it

File: test/Redact/ClientTest.php [hunk 1]
```diff
@@ -85,11 +85,10 @@ public function testOptionsDoNotOverwriteParams()
 
     /**
      * @dataProvider exceptionsProvider
+     * @expectedException Nexmo\Client\Exception\Request
```
Comment (mheap) [inline test/Redact/ClientTest.php]: We lose a test by moving this to an annotation. `setExpectedException` is deprecated but we can use `$this->expectException($expectedException);` and `$this->expectExceptionMessage ($expectedMessage);` instead

Comment (lornajane) [reply test/Redact/ClientTest.php]: Thanks! I added this

**Comment by lornajane** [comment]
We're stopping here: PHPUnit v8 is out but requires PHP 7.2, we're minimum version PHP 7.1 so let's stick on PHPUnit 7.4+ but <8.0 for now.

**Comment by lornajane** [comment]
Also fixes #168 as a happy side effect

**Pull request merged**